src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,52 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private String createReport(List<String> dataFromFile) {
+        int supplies = 0;
+        int buys = 0;
+        int result;
+        for (String content : dataFromFile) {
+            String action = content.split(",")[0];
+            String sum = content.split(",")[1];
+            if (action.equals("supply")) {
+                supplies += Integer.parseInt(sum);
+            }
+            if (action.equals("buy")) {
```

Comment:
extract actions to constants. and it's better to check like "supply".equals(action) since action can be null and you may catch NullPointerException with first approach

Reply:
Thanks! Thats a good one.